File: 0150-evaluate-reverse-polish-notation/0150-evaluate-reverse-polish-notation.java
class Solution {
    public int evalRPN(String[] tokens) {
        Stack<Integer> st = new Stack<>();
        int ans = 0;
        int a,b;
        for(String s:tokens){
            if(s.equals("+")){
                st.push(st.pop()+st.pop());
            }else if(s.equals("-")){
                b = st.pop();
                a = st.pop();
                st.push(a-b);
            }else if(s.equals("*")){
                st.push(st.pop()*st.pop());
            }else if(s.equals("/")){
                b = st.pop();
                a = st.pop();
                st.push(a/b);
            }else{
                st.push(Integer.parseInt(s));
            }
        }
        return st.pop();
    }
}

File: 0150-evaluate-reverse-polish-notation/0150-evaluate-reverse-polish-notation.py
class Solution:
    def evalRPN(self, tokens: List[str]) -> int:
        st = []
        for token in tokens:
            if token == "+":
                st.append(st.pop()+st.pop())
            elif token == "*":
                st.append(st.pop()*st.pop())
            elif token == "-":
                a = st.pop()
                b = st.pop()
                st.append(b-a)
            elif token == "/":
                a = st.pop()
                b = st.pop()
                st.append(int(b/a))
            else:
                st.append(int(token))
        return st.pop()